src/error.rs

```diff
@@ -12,20 +12,25 @@ use core::fmt;
 #[cfg(not(target_env = "sgx"))]
 use std::{io, error};
 
+// A randomly-chosen 16-bit prefix for our codes
+pub(crate) const CODE_PREFIX: u32 = 0x57f40000;
+const CODE_UNKNOWN: u32 = CODE_PREFIX | 0;
+const CODE_UNAVAILABLE: u32 = CODE_PREFIX | 1;
```

Comment:
Maybe we should make `CODE_UNKNOWN` and `CODE_UNAVAILABLE` associated constants of `Error` type?